src/lem/multiframe.rs

```diff
@@ -48,7 +48,7 @@ pub struct MultiFrame<'a, F: LurkField, C: Coprocessor<F>> {
     input: Option<Vec<Ptr>>,
     output: Option<Vec<Ptr>>,
     frames: Option<Vec<Frame>>,
-    cached_witness: Option<WitnessCS<F>>,
+    cached_witness: Option<(WitnessCS<F>, Vec<AllocatedNum<F>>)>,
```

Comment:
Couldn't this be a `OnceCell`? It would make it easy to ensure `cache_witness` is idempotent.

Reply:
Good idea! Will change it